src/compass/utils/h5_helpers.py

```diff
@@ -79,6 +79,33 @@ def add_dataset_and_attrs(group, meta_item):
         val_ds.attrs[key] = _as_np_string_if_needed(val)
 
 
+def _determine_fill_value(dtype):
+    '''
+    Helper function to determine COMPASS specific fill values based on h5py
+    Dataset type (dtype)
+    '''
+    # Possible float types and float fill
+    float_types = [np.double, np.single, np.float32, np.float64, 'float32',
+                   'float64']
+    float_fill = np.nan
+
+    # Possible complex types and complex fill
+    complex_types = [np.complex128, 'complex64', np.complex64, 'complex32']
+    complex_fill = np.nan * (0 + 1j)
```

Comment:
Since this is a tunable parameter in the core module, I am wondering if we can extract it in a more automated way (just in case we change decision on what would be the fill value). Same for other data type. Also, have we made sure that we have been consistent for datasets with the same data type? Please, double-check.

Reply:
To address your concerns, I've created a new file (`fill_value.py`) that includes:
1. default fill values in a `dataclass`
2. said `dataclass` can handle user defined values
3. `determine_fill_value` function (moved from `h5_helpers.py`)

I'll test more robustly over the weekend.